entry-generation/godot-annotation-processor/src/main/kotlin/godot/annotation/processor/GodotAnnotationProcessor.kt

```diff
@@ -0,0 +1,106 @@
+    private val signals: MutableSet<PropertyDescriptor> = mutableSetOf()
+
+    override fun process(roundEnvironment: RoundEnvironment) {
+        classes.addAll(
+            roundEnvironment
+                .getElementsAnnotatedWith("godot.annotation.RegisterClass")
+                .map { it as Element.ClassElement }
+                .map { it.classDescriptor }
+        )
+
+        properties.addAll(
+            roundEnvironment
+                .getElementsAnnotatedWith("godot.annotation.RegisterProperty")
+                .map { it as Element.PropertyElement }
+                .map { it.propertyDescriptor }
+        )
+
+        functions.addAll(
+            roundEnvironment
+                .getElementsAnnotatedWith("godot.annotation.RegisterFunction")
+                .map { it as Element.FunctionElement }
+                .map { it.func }
+        )
+
+        signals.addAll(
+            roundEnvironment
+                .getElementsAnnotatedWith("godot.annotation.RegisterSignal")
+                .map { it as Element.PropertyElement }
+                .map { it.propertyDescriptor }
+        )
+
+        performSanityChecks()
+    }
+
+    private fun performSanityChecks() {
+        classes.forEach {
+            if (it.constructors.size > 1) {
+                throw IllegalClassFormatException("A Class annotated with \"@RegisterClass\" can only have a default constructor!\nBut ${it.name} contains ${it.constructors.size} constructors")
+            }
+        }
+        functions.forEach {
+            if (!classes.contains(it.containingDeclaration)) {
+                throw Exception("${it.containingDeclaration.name.asString()} contains a registered function: ${it.name} but is not annotated with @RegisterClass! Classes containing functions which are registered, also have to be registered!")
+            }
+        }
+        properties.forEach {
+            if (!classes.contains(it.containingDeclaration)) {
+                throw Exception("${it.containingDeclaration.name.asString()} contains a registered property: ${it.name} but is not annotated with @RegisterClass! Classes containing properties which are registered, also have to be registered!")
+            }
+        }
+        signals.forEach {
+            if (!classes.contains(it.containingDeclaration)) {
+                throw Exception("${it.containingDeclaration.name.asString()} contains a signal: ${it.name} but is not annotated with @RegisterClass! Classes containing signals, also have to be registered!")
+            }
+        }
+    }
+
+    override fun processingOver() {
+        //TODO: entry generation
+    }
+
+    /**
+     * Returns the path as a String to which the generated code should be written
+     * @return directory path to where the generated code should be written to as a String
+     */
+    private fun getKaptGeneratedDirectory(): String {
+        return "${Paths.get("").toAbsolutePath()}/build/godot/entries"
```

Comment:
This value can be passed from gradle to the compiler plugin via the subplugin, can we make this a property of `GodotAnnotationProcessor`?

Reply:
Sure. Good idea.